### PR TITLE
Slightly changes IsBanned() to include something TG does

### PR DESCRIPTION
**Pull request by ShiftyRail**
Again, we're kinda shooting birds at night here, but I've noticed TG had this in their code for at least 4 years, and unlike us, they never had this problem of connections being silently dropped. What's the worst that could happen ?

~~Also includes getting rid of the space in file names for client_procs and client_defines because it's silly~~

### CODE REVIEW

**Comment by wotisjanitor** [comment]
>to include something TG

Stopped reading right there.

File: code/modules/admin/IsBanned.dm [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /world/IsBanned(key, address, computer_id, type)
 	var/real_login = type != "goonchat" //Certain actions don't make sense to perform for the cookie checks.
 
+	// Some "world-type" bans relegated to BYOND's internal cuisine. That's how TG does it.
+	if (type == "world")
+		return ..()
```
Comment (boy2mantwicethefam) [inline code/modules/admin/IsBanned.dm:6]: What does this do?

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -127,6 +127,10 @@
 	///////////
 /client/New(TopicData)
 	// world.log << "creating chatOutput"
+
+	var/tdata = TopicData //save this for later use
```
Comment (boy2mantwicethefam) [inline code/modules/client/client_procs.dm]: This isn't used anywhere else

File: code/modules/admin/IsBanned.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/modules/admin/IsBanned.dm:6]: This defers the banned check to byond's internal proc for some kind of connections. In this case, it's for when you've been redirected through World.Topic().

```dm
	if (type == "world")
		return ..() //shunt world topic banchecks to purely to byond's internal ban system
```

TG seems to do it just fine.

File: code/modules/client/client_procs.dm (hunk 2 above)
Comment (ShiftyRail) [reply code/modules/client/client_procs.dm]: ye

**Comment by ShiftyRail** [comment]
removed the unused stuff, picked up something else that's interesting:

```
	var/client/C = directory[ckey]
	if (C && ckey == C.ckey && computer_id == C.computer_id && address == C.address)
		return //don't recheck connected clients.
```

Maybe it helps ? I don't know, we're basically shooting birds at night.

**Comment by Securitymain** [comment]
What is this supposed to do?

**Comment by DorainGrey** [comment]
This is interesting, as I recall isbanned() was prime suspect for connection dropping when it was being debugged.
Definitely worth a try.

**Comment by DorainGrey** [comment]
> 	var/client/C = directory[ckey]
> 	if (C && ckey == C.ckey && computer_id == C.computer_id && address == C.address)
> 		return //don't recheck connected clients.
> ```
> 
> Maybe it helps ? I don't know, we're basically shooting birds at night.

In what scenario, does this this happen I wonder? Surely isbanned() is only called once?
On byond reconnect?

I've seen bans be applied, that take till next round to take affect.

**Comment by Kurfursten** [comment]
Have we really held this up 3 weeks over what appear to be empty changes to secret repo files?

Please just handle the secret repo conflict yourself and self-merge after.

**Review by Kurfursten — APPROVED** [review]

**Comment by ShiftyRail** [comment]
turns out we don't need it, we found the real issue.

**Pull request closed without merging**